src/sweeper/report.py

```diff
@@ -8,58 +8,65 @@
 import os
 from datetime import datetime
 
+def _generate_report(writer, reports):
+    if len(reports) == 0:
+        writer('No issues found!')
 
-def save_report(data_dict, title, fc, path):
-    now = datetime.now().strftime('%Y%m%d_%H%M%S_%f')[:-3]
-    name = f'{title}_report_{now}.txt'
+        return None
 
-    filename = os.path.join(path, name)
+    #: loop through each report dict in the list of dicts
+    for report in reports:
+        writer(f'{report["title"]} Report for {report["feature_class"]}')
+        issues_found = len(report['issues'])
+        if issues_found == 0:
+            writer('No issues found!')
+            writer('---')
 
-    with open(filename, 'w') as textfile:
-        textfile.write(f"{title.title()} Report for '{fc}' \n \n")
+            continue
 
-        if len(data_dict) == 0:
-            textfile.write('No issues found!')
+        writer(f'{issues_found} issues found')
+        writer('  Issues found:')
 
-            return filename
+        for oid in report['issues']:
+            writer(f'    ObjectID {oid}')
 
-        lines = []
-        textfile.write(f'{len(data_dict)} issues found in feature class \n \n')
-        textfile.write('Issues found: \n')
+        writer('\nSelect statement to view issues in ArcGIS:')
+        statement = f'OBJECTID IN ({",".join(report["issues"])})'
 
-        for key, value in data_dict.items():
-            lines.append(f'ObjectID {key}: {value} \n')
+        writer(statement)
 
-        textfile.writelines(lines)
-        textfile.write('\n')
-        textfile.write('Select statement to view issues in ArcGIS: \n')
+        writer('---')
 
-        statement = f'OBJECTID IN ({",".join(str(key) for key in data_dict)})'
+def save_report(reports, save_directory):
+    '''
+    save_directory 
+      - folder sweeper-run-{date}
+        - feature class name-sweeper name-numberofissues.txt `Counties-Empties-5.txt`   `Counties-Empties-0.txt`
+    '''
+    report_directory = _create_report_directory(save_directory)
 
-        textfile.write(statement)
+    for report in reports:
+        file_name = _get_file_name(report)
+        file_path = os.path.join(report_directory, file_name)
 
-        return filename
+        with open(file_path, 'w') as textfile:
+            def write_lines(text):
+                textfile.writelines(f'{text}\n')
 
-def print_report(data_dict, title, fc):
-    report = f"{title.title()} Report for '{fc}' \n"
+            _generate_report(write_lines, [report])
```

Comment:
[52-56] I guess I'm a bit confused with what's going on here.  We create a function 'write_lines' that needs a 'text' parameter.  But in line 56, '_generate_report(write_lines, [report])', we don't seem to pass in an argument.

I guess the argument is populated within the '_generate_report' function with each call of writer(xxxxxxx)?

Reply:
Nice job on the multiline comment! That's a new github feature.

You are absolutely correct. We are passing a function as a parameter to another function. We had to add the extra inline function bit because `textfile.write` does not append a new line like the `print` statement does (so everything was being appended to the same line), and `textfile.writelines` expects an array (when print expects a string). So we had to get a little fancy and modify the input parameter when writing to a file.

There is a bit of a bug though on line 54. the `\n` is extra and is an artifact of a refactor.

Reply:
thanks.  I removed the `\n' from line 54.